src/Application/Actions/Donations/Confirm.php

```diff
@@ -92,7 +93,33 @@ protected function action(Request $request, Response $response, array $args): Re
         }
 
         $paymentIntentId = $donation->getTransactionId();
-        $paymentMethod = $this->stripe->retrievePaymentMethod($paymentMethodId);
+        try {
+            $paymentMethod = $this->stripe->updatePaymentMethodBillingDetail($paymentMethodId, $donation);
+        } catch (CardException $cardException) {
+            return $this->handleCardException(
+                context: 'updatePaymentMethodBillingDetail',
+                exception: $cardException,
+                donation: $donation,
+                paymentIntentId: $paymentIntentId,
+            );
+        } catch (ApiErrorException $exception) {
+            $this->logger->error(sprintf(
+                'Stripe %s on Confirm updatePaymentMethodBillingDetail for donation %s (%s): %s',
+                get_class($exception),
+                $donation->getUuid(),
+                $paymentIntentId,
+                $exception->getMessage(),
+            ));
+
+            $this->entityManager->rollback();
+
+            return new JsonResponse([
+                'error' => [
+                    'message' => $exception->getMessage(),
+                    'code' => $exception->getStripeCode(),
+                ],
+            ], 500);
```

Comment:
Logging the detail is useful, but instead of rolling back and returning we could probably just re-throw the exception, which seems a bit simpler and afaik effectively does the same thing. Not a blocker.

Reply:
https://github.com/thebiggive/matchbot/pull/726/commits/3ab0d9aa83521c10db85209839a48587ba9b1d95